internal/controller/dnsrecord_controller.go

```diff
@@ -167,28 +174,28 @@ func (r *DNSRecordReconciler) deleteRecord(ctx context.Context, dnsRecord *v1alp
 	err := r.Get(ctx, client.ObjectKeyFromObject(managedZone), managedZone, &client.GetOptions{})
 	if err != nil {
 		// If the Managed Zone isn't found, just continue
-		return client.IgnoreNotFound(err)
+		return noRequeueDuration, client.IgnoreNotFound(err)
 	}
 	managedZoneReady := meta.IsStatusConditionTrue(managedZone.Status.Conditions, "Ready")
 
 	if !managedZoneReady {
-		return fmt.Errorf("the managed zone is not in a ready state : %s", managedZone.Name)
+		return validationRequeueTime, fmt.Errorf("the managed zone is not in a ready state : %s", managedZone.Name)
 	}
 
-	_, err = r.applyChanges(ctx, dnsRecord, managedZone, true)
+	requeueTime, err := r.applyChanges(ctx, dnsRecord, managedZone, true)
 	if err != nil {
 		if strings.Contains(err.Error(), "was not found") || strings.Contains(err.Error(), "notFound") {
 			logger.Info("Record not found in managed zone, continuing", "dnsRecord", dnsRecord.Name, "managedZone", managedZone.Name)
-			return nil
+			return noRequeueDuration, nil
 		} else if strings.Contains(err.Error(), "no endpoints") {
 			logger.Info("DNS record had no endpoint, continuing", "dnsRecord", dnsRecord.Name, "managedZone", managedZone.Name)
-			return nil
+			return noRequeueDuration, nil
 		}
-		return err
+		return noRequeueDuration, err
```

Comment:
does `noRequeueDuration` have the effect as returning `ctrl.Result{}, updateError`?

Reply:
yes. The way it is interpreted: if `err != nil` it will ignore `ctrl.Result{}` entirely and will requeue immediately. If `err == nil` then it will look at instructions for requeue. 
This makes me think that I might made it more confusing than it should be. There was a suggestion to just roam with the boolean flag saying if there were changes published or not and we decide on how to requeue in the very last step. I'm planning on having that implemented after the code freeze (but if that will make more sense could have it ready by the end of a day)

Reply:
no leave as is for now and can follow up with improvements